### PR TITLE
ROX-20754: Onboard Operator build to RHTAP

### PR DESCRIPTION
**Pull request by tommartensen**
## Description

This build is very similar to roxctl (https://github.com/stackrox/stackrox/pull/7982).
Enterprise contracts are broken on RHTAP due to pending Snyk integration, see https://issues.redhat.com/browse/ROX-20237.  

Dockerfile based on https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/blob/rhacs-1.0-rhel-8/distgit/containers/rhacs-operator/Dockerfile.in.

## Checklist
- ~[ ] Investigated and inspected CI test results~
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

- Build completes in RHTAP. 
- Verified container starts, image: `quay.io/redhat-user-workloads/rh-acs-tenant/acs/operator:on-pr-9b5594be0bd24c24b8d3de19bc2ab351d18c6881`

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: operator/rhtap.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+LABEL \
+    com.redhat.component="rhacs-operator-container" \
+    com.redhat.license_terms="https://www.redhat.com/agreements" \
+    description="Operator for Red Hat Advanced Cluster Security for Kubernetes" \
+    io.k8s.description="Operator for Red Hat Advanced Cluster Security for Kubernetes" \
+    io.k8s.display-name="operator" \
+    io.openshift.tags="rhacs,operator,stackrox" \
+    maintainer="Red Hat, Inc." \
+    name="rhacs-operator-rhel8" \
```
Comment (kylape) [inline operator/rhtap.Dockerfile:29]: not sure if this matters, but this differs from what's in the midstram dockerfile (`rhacs-rhel8-operator`).

**Review by kylape — COMMENTED** [review]
Looks reasonable to me.  For some reason I couldn't review the build in RHTAP -- it seemed to have disappeared.  (All the build components were in a weird state, like all their history had been wiped.)

File: .tekton/operator-pull-request.yaml [hunk 2]
```diff
@@ -0,0 +1,437 @@
...
+        operator: in
+        values: [ "true" ]
+      - input: $(params.build-source-image)
+        operator: in
+        values: [ "true" ]
+      workspaces:
+      - name: workspace
+        workspace: workspace
+
+    - name: inspect-image
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/operator-pull-request.yaml:305]: Good catch, updated the push pipeline to match the PR pipeline.

File: .tekton/operator-push.yaml [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+  - name: workspace
+    volumeClaimTemplate:
+      metadata:
+        creationTimestamp: null
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 1Gi
```
[the remark this answers is not in the document]
Comment (tommartensen) [reply .tekton/operator-push.yaml]: Good catch, updated the push pipeline to match the PR pipeline.

File: operator/rhtap.Dockerfile (hunk 1 above)
Comment (tommartensen) [reply operator/rhtap.Dockerfile]: Updated it to midstream `name` label. 
There is a difference between `operator` and `roxctl` name label formatting:
- operator: `rhacs-rhel8-operator` (https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/blob/rhacs-1.0-rhel-8/distgit/containers/rhacs-operator/Dockerfile.in#L24) 
- roxctl: `rhacs-roxctl-rhel8` (https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/blob/rhacs-1.0-rhel-8/distgit/containers/rhacs-roxctl/Dockerfile.in#L26)

I think it's better to keep it like that.

**Review by kylape — APPROVED** [review]
LGTM!

**Pull request merged**